sympy/matrices/common.py

```diff
@@ -2224,6 +2229,9 @@ def __neg__(self):
 
     @call_highest_priority('__rpow__')
     def __pow__(self, exp):
+        return self.pow(exp, mulsimp=False)
+
+    def pow(self, exp, mulsimp=True):
```

Comment:
It would also be good if `pow` would give the option to choose between recursion or Jordan form for calculating the power.

Reply:
For that you could just directly call `_matrix_pow_by_jordan_blocks`

Reply:
The point is to make these different options available publicly and to provide a place to document them and make them discoverable by users.

Reply:
I added info to matrix.rst and intend to add benchmarks, what else do you propose?

Reply:
When I suggested a pow method I didn't mean for it to be an alternative to using `__pow__` which is implicitly how you've represented it in matrix.rst. What I meant is that the `pow` method would be an API entry point allowing users to choose between all the different matrix power routines available. The default flags to `pow` would give the same behaviour as `__pow__` but using `pow` enables customising the result like being able to explicitly choose to use the Jordan form method even if the exponent is an Integer.

Reply:
I can add that sure.

Reply:
So u want `mul()` to default to no simplification as well?

Reply:
Yes I would think that mul defaults to no simplification. Then we can explicitly document that the default behaviour of mul is the same as `__mul__`.